crowbar_framework/app/models/service_object.rb

```diff
@@ -1418,6 +1440,48 @@ def apply_role(role, inst, in_queue)
     [200, {}]
   end
 
+  def delete_remove_role_from_runlist(nodes_with_roles_to_remove)
```

Comment:
This method is incorrectly named - the name implies that it only removes one role from one runlist, but actually it will remove multiple roles from multiple runlists.  Better to call it `delete_remove_roles_from_runlists` and have the body of the inner loop call `delete_remove_role_from_runlist`.